tests/utilities/test_cli.py

```diff
@@ -329,3 +331,93 @@ def test_lightning_cli_config_and_subclass_mode(tmpdir):
     assert config['model'] == cli.config['model']
     assert config['data'] == cli.config['data']
     assert config['trainer'] == cli.config['trainer']
+
+
+def any_model_any_data_cli():
+    LightningCLI(
+        LightningModule,
+        LightningDataModule,
+        subclass_mode_model=True,
+        subclass_mode_data=True,
+    )
+
+
+def test_lightning_cli_help():
+
+    cli_args = ['any.py', '--help']
+    out = StringIO()
+    with mock.patch('sys.argv', cli_args), redirect_stdout(out), pytest.raises(SystemExit):
+        any_model_any_data_cli()
+
+    assert '--print_config' in out.getvalue()
+    assert '--config' in out.getvalue()
+    assert '--seed_everything' in out.getvalue()
```

Comment:
is this correct? what does this seed_everything do here? It's a function not a cli arg.

Reply:
This is for reproducibility. If `seed_everything` is set, then the `seed_everything` function is called with this value before instatiation of classes. And the seed value will be included in the config file saved to `lightning_logs` along with the trainer, model and data settings. Thus if you run the cli again with the same config it will get the same seed value.